File: app/src/main/java/com/example/architectureapplication/di/AddNoteComponent.java
package com.example.architectureapplication.di;

import com.example.architectureapplication.AddNoteActivityKt;

import dagger.Subcomponent;

@Subcomponent()
public interface AddNoteComponent {

    @Subcomponent.Builder
    interface Builder{
        AddNoteComponent build();
    }

    void inject(AddNoteActivityKt addNoteActivity);
}

File: app/src/main/java/com/example/architectureapplication/kt/AddNoteComponent.kt
package com.example.architectureapplication.kt

import com.example.architectureapplication.AddNoteActivity
import dagger.Subcomponent

@Subcomponent
interface AddNoteComponent {
    fun inject(activityKt: AddNoteActivity)

    @Subcomponent.Builder
    interface Builder{
        fun build(): AddNoteComponent
    }
}
